eppo_core/src/configuration_poller.rs

```diff
@@ -0,0 +1,188 @@
+use std::{sync::Arc, time::Duration};
+
+use rand::{thread_rng, Rng};
+use tokio::sync::watch;
+use tokio_util::sync::CancellationToken;
+
+use crate::{
+    configuration_fetcher::ConfigurationFetcher, configuration_store::ConfigurationStore, Error,
+};
+
+/// Configuration for [`configuration_poller`].
+// Not implementing `Copy` as we may add non-copyable fields in the future.
+#[derive(Debug, Clone)]
+pub struct ConfigurationPollerConfig {
+    /// Interval to wait between requests for configuration.
+    ///
+    /// Defaults to [`ConfigurationPollerConfig::DEFAULT_POLL_INTERVAL`].
+    pub interval: Duration,
+    /// Jitter applies a randomized duration to wait between requests for configuration. This helps
+    /// to avoid multiple server instances synchronizing and producing spiky network load.
+    ///
+    /// Defaults to [`ConfigurationPollerConfig::DEFAULT_POLL_JITTER`].
+    pub jitter: Duration,
+}
+
+impl ConfigurationPollerConfig {
+    /// Default value for [`ConfigurationPollerConfig::interval`].
+    pub const DEFAULT_POLL_INTERVAL: Duration = Duration::from_secs(30);
+    /// Default value for [`ConfigurationPollerConfig::jitter`].
+    pub const DEFAULT_POLL_JITTER: Duration = Duration::from_secs(3);
+
+    /// Create a new `ConfigurationPollerConfig` using default configuration.
+    pub fn new() -> ConfigurationPollerConfig {
+        ConfigurationPollerConfig::default()
+    }
+
+    /// Update poll interval with `interval`.
+    pub fn with_interval(mut self, interval: Duration) -> ConfigurationPollerConfig {
+        self.interval = interval;
+        self
+    }
+
+    /// Update poll interval jitter with `jitter`.
+    pub fn with_jitter(mut self, jitter: Duration) -> ConfigurationPollerConfig {
+        self.jitter = jitter;
+        self
+    }
+}
+
+impl Default for ConfigurationPollerConfig {
+    fn default() -> ConfigurationPollerConfig {
+        ConfigurationPollerConfig {
+            interval: ConfigurationPollerConfig::DEFAULT_POLL_INTERVAL,
+            jitter: ConfigurationPollerConfig::DEFAULT_POLL_JITTER,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct ConfigurationPoller {
+    status: watch::Receiver<Option<Result<(), crate::Error>>>,
+    cancellation_token: CancellationToken,
```

Comment:
Cancellation token is stored here if we ever want to stop the configuration poller without stopping the whole background runtime.